src/OXRS_WT32.cpp

```diff
@@ -512,14 +553,96 @@ void OXRS_WT32::_initialiseRestApi(void)
 
   // Set up the REST API
   _api.begin();
-  
+
   // Register our callbacks
   _api.onAdopt(_apiAdopt);
-  
+
   // Start listening
   _server.begin();
 }
 
+void OXRS_WT32::_initialiseClimateSensor(void)
+{
+  // Initialise the onboard SHT20 climate sensor
+  sht.begin(I2C0_SDA, I2C0_SCL);
+  delay(10);
+
+  _climateSensorFound = sht.isConnected();
+
+  if (!_climateSensorFound)
+  {
+    _logger.println(F("[wt32] no SHT20 sensor found"));
+    return;
+  }
+
+  _lastClimateUpdate = -_climateUpdateMs;
+}
+
+// get values from climate sensor, store local, publish /tele
+void OXRS_WT32::_updateClimateSensor(void)
+{
+  // Ignore if climate sensor not found or has been disabled
+  if (!_climateSensorFound || _climateUpdateMs == 0)
+  {
+    return;
+  }
+
+  // Check if we need to get new readings and publish
+  if ((millis() - _lastClimateUpdate) > _climateUpdateMs)
+  {
+    // Read values from onboard sensor
+    sht.read();
+
+    int temperature = round(sht.getTemperature() * 10);
+    int humidity = round(sht.getHumidity() * 10);
+    _temperature = (double)temperature / 10.0;
+    _humidity = (double)humidity / 10.0;
+
+    // Publish climate to mqtt
+    StaticJsonDocument<64> json;
+    json["temperature"] = _temperature;
+    json["humidity"] = _humidity;
+    publishTelemetry(json.as<JsonVariant>());
+
+    // Reset our timer
+    _lastClimateUpdate = millis();
+    _climateUpdated = true;
+  }
+}
+
+// get climate sensor values
+bool OXRS_WT32::getClimate(float *temperature, float *humidity)
+{
+  *temperature = _temperature;
+  *humidity = _humidity;
+
+  _climateUpdated = false;
+  return (isnan(_temperature) || isnan(_humidity)) == false;
+}
+
+// return the update status
+bool OXRS_WT32::getClimateUpdated(void)
```

Comment:
Not sure I follow why you need this and all the associated state variables - e.g. `_climateUpdateIntervallChanged` and `_climateUpdated` etc.

Reply:
I'm also not happy with that. The reason is to synchronize the settings screen with recent data even if value changes during displaying the settings screen and reacting immediately when climate measurements are turned on / off. We discussed that with rack32 in the past. 
I think the more straight forward solution will be to use a callback into main that can be used by the WT32 lib to update the settings screen direkt. (push instead of pull). 
I think I have a plan.